DOCSP-40410 get started (#33)
mongodb/docs-java-rs: source/get-started/connect-to-mongodb.txt

.. _java-rs-get-started-connect-to-mongodb:

==================
Connect to MongoDB
==================

.. TODO: 
.. Add this sentence after the last sentence in the paragraph for implementing reactive streams subscribers. 
.. To learn more about the Java Reactive Streams API, see link here.

.. procedure::
   :style: connected

    .. step:: Implement Reactive Streams Subscribers

      Create a folder within the Java folder in your Maven or Gradle project named ``helpers``.
      In the ``helpers`` folder, create a Java file named ``SubscriberHelpers``.
      The Java Reactive Streams driver is an implementation of the Reactive
      Streams API, which consists of three components: Publisher, Subscriber,
      and Subscription. 
      
      The following code is an implementation of six

>       different ``Subscriber`` types used in your {+driver-short+} application.

Author: The query actually uses multiple subscribers, and I was only able to take out one, as a lot of them are extensions of each other (so I couldn't take out ConsumerSubscriber because I needed it for PrintDocumentSubscriber

>       different ``Subscriber`` types used in your {+driver-short+} application.

Reviewer: Requesting feedback from the tech reviewer: is there a way that we can condense and simplify the Subscribers for the purposes of the tutorial?

      .. literalinclude:: /includes/get-started/subscriber-helpers.java
         :language: java
         :copyable:

   .. step:: Create your {+driver-short+} Application
      
      Create a folder within the Java folder in your Maven or Gradle project named ``org.example``.
      In the ``org.example`` folder, create a Java file named ``QuickStart``. Copy and paste the following code 
      into the ``QuickStart`` file:
      
      .. literalinclude:: /includes/get-started/connect-and-ping.java
         :language: java
         :copyable:

   .. step:: Assign the Connection String

      Replace the ``<connection string URI>`` placeholder with the 
      connection string that you copied from the :ref:`java-rs-get-started-connection-string`
      step of this guide.

   .. step:: Run your Application

      Run your application, either in your IDE or in your shell. The output
      shows that you have connected to MongoDB.

      .. code-block:: none

        Pinged your deployment. You successfully connected to MongoDB!

      .. tip::

         If you encounter an error, check whether you specified the
         proper connection string. The ``Logging is disabled`` error 

>          is standard and does not block your query.

Author: I'm worried that if we do not document it, people will take that error as an actual error and think that means they did the tutorial wrong. Standard might not be the right word--my thinking was to tell users that the error is not blocking/does not mean they did anything wrong?

.. _java-rs-get-started-connect-to-mongodb:

==================
Connect to MongoDB
==================

.. TODO: 
.. Add this sentence after the last sentence in the paragraph for implementing reactive streams subscribers. 
.. To learn more about the Java Reactive Streams API, see link here.

> The Java Reactive Streams driver is an implementation of the Reactive Streams API, which consists of three components: 

Reviewer: ```suggestion
The {+driver-short+} is an implementation of the Reactive Streams API, which consists of three components: 
```

- Publisher
- Subscriber
- Subscription

> In this tutorial, you must implement Java Reactive Streams Subscribers to
> query your Atlas database.

Reviewer: Replace with source constant. I also think we should remove "Atlas" and just refer to it as their database since it's not necessarily an _Atlas database_, but a database hosted on Atlas

```suggestion
In this tutorial, you must implement {+java-rs+} Subscribers to
query your database.
```

of ``Subscriber``. A ``Subscription`` represents a one-to-one lifecycle of a ``Subscriber`` subscribing to a ``Publisher``.

.. tip::

   To learn more about reactive streams, visit
   the `Reactive Streams documentation <http://www.reactive-streams.org>`__.

In this tutorial, you must implement {+java-rs+} ``Subscribers`` to
query your database. You must also use an integrated development environment
(IDE) to complete the tutorial. This guide uses methods from the Reactor library, a library
based on the Reactive Streams specification to implement {+java-rs+}

> ``Subscribers``. 

Reviewer 2: S: Right now, this paragraph is describing a lot of different aspects of the sample app. Can you organize this information into the steps where it is relevant or restructure the paragraph to be more cohesive, perhaps as a list of items to keep in mind or something?

      Copy and paste the following code 
      into the ``SubscriberHelpers`` file:
      
      .. literalinclude:: /includes/get-started/subscriber-helpers.java
         :language: java
         :copyable:

   .. step:: Create your {+driver-short+} Application
      
      In your project, create a new Java file in the ``Java`` package named 

>       ``QueryDatabase`` . Copy and paste the following code into the ``QueryDatabase`` file:

Reviewer: ```suggestion
      ``QueryDatabase``. Copy and paste the following code into the ``QueryDatabase`` file:
```

      
      .. literalinclude:: /includes/get-started/query-database.java
         :language: java
         :copyable:

   .. step:: Assign the Connection String

      Replace the ``<connection string>`` placeholder with the 
      connection string that you copied from the :ref:`java-rs-get-started-connection-string`
      step of this guide.

   .. step:: Run your Application

      Run your application, either in your IDE or in your shell. The output
      shows that you have connected to MongoDB and queried the database.

      .. code-block:: none

        {
           _id: ...,
           plot: 'A young man is accidentally sent 30 years into the past...',
           genres: [ 'Adventure', 'Comedy', 'Sci-Fi' ],
           ...
           title: 'Back to the Future',
           ...
         }

> 

Reviewer 2: S: remove extra lines
